### PR TITLE
Rustfmt-ing libarena.

### PR DESCRIPTION
**Pull request by goyox86**
Hi Rustaceans!

This is the result of running latest rustfmt on libarena!

//cc @nrc

### CODE REVIEW

File: src/libarena/lib.rs [hunk 1]
```diff
@@ -197,10 +197,11 @@ fn un_bitpack_tydesc_ptr(p: usize) -> (*const TyDesc, bool) {
 struct TyDesc {
     drop_glue: fn(*const i8),
     size: usize,
-    align: usize
+    align: usize,
 }
 
-trait AllTypes { fn dummy(&self) { } }
+trait AllTypes {    fn dummy(&self) {
+    } }
```
Comment (oli-obk) [inline src/libarena/lib.rs]: this is not an improvement. I'd report a bug against rustfmt

File: src/libarena/lib.rs [hunk 2]
```diff
@@ -297,24 +293,25 @@ impl<'longer_than_self> Arena<'longer_than_self> {
 
         unsafe {
             let buf = head.as_ptr();
-            (buf.offset(tydesc_start as isize), buf.offset(start as isize))
+            (buf.offset(tydesc_start as isize),
+             buf.offset(start as isize))
```
Comment (oli-obk) [inline src/libarena/lib.rs]: why does this get split into two lines when there are much longer lines in the file?

File: src/libarena/lib.rs [hunk 3]
```diff
@@ -393,9 +392,7 @@ struct TypedArenaChunk<T> {
     next: *mut TypedArenaChunk<T>,
 
     /// The number of elements that this chunk can hold.
-    capacity: usize,
-
-    // Objects follow here, suitably aligned.
+    capacity: usize, // Objects follow here, suitably aligned.
```
Comment (oli-obk) [inline src/libarena/lib.rs]: this comment placement is not correct

File: src/libarena/lib.rs [hunk 4]
```diff
@@ -572,8 +576,9 @@ mod tests {
 
         let arena = Wrap(TypedArena::new());
 
-        let result = arena.alloc_outer(|| Outer {
-            inner: arena.alloc_inner(|| Inner { value: 10 }) });
+        let result = arena.alloc_outer(|| {
```
Comment (oli-obk) [inline src/libarena/lib.rs]: an additional block was generated here. Is this the expected behavior?

File: src/libarena/lib.rs (hunk 1 above)
Comment (nrc) [inline src/libarena/lib.rs]: yeah, yuck

File: src/libarena/lib.rs (hunk 2 above)
Comment (nrc) [inline src/libarena/lib.rs]: https://github.com/nrc/rustfmt/issues/450

File: src/libarena/lib.rs (hunk 3 above)
Comment (nrc) [inline src/libarena/lib.rs]: This probably needs to have a `#[rustfmt_skip]`, I don't think we'll ever cope with this

Comment (gkoz) [inline src/libarena/lib.rs]: Is there an issue for this? Mangling any commented out lines at the bottom of a struct might be a bummer in some cases.

Comment (nrc) [inline src/libarena/lib.rs]: not yet

Comment (nrc) [inline src/libarena/lib.rs]: https://github.com/nrc/rustfmt/issues/457

Comment (gkoz) [inline src/libarena/lib.rs]: Thanks!

File: src/libarena/lib.rs (hunk 1 above)
Comment (goyox86) [inline src/libarena/lib.rs]: https://github.com/nrc/rustfmt/issues/460

**Comment by goyox86** [comment]
@nrc Manual fixups done as well as reported rustfmt issues.

File: src/libarena/lib.rs [hunk 5]
```diff
@@ -105,7 +105,7 @@ pub struct Arena<'longer_than_self> {
     head: RefCell<Chunk>,
     copy_head: RefCell<Chunk>,
     chunks: RefCell<Vec<Chunk>>,
-    _marker: marker::PhantomData<*mut &'longer_than_self()>,
+    _marker: marker::PhantomData<*mut &'longer_than_self ()>,
```
Comment (nrc) [inline src/libarena/lib.rs]: hmm, this seems to be erroneous too on second glance

Comment (nrc) [inline src/libarena/lib.rs]: https://github.com/nrc/rustfmt/issues/467

Comment (nrc) [inline src/libarena/lib.rs]: Oh actually, it's not sorry, that's a lifetime

**Comment by goyox86** [comment]
@nrc Comment fixed

**Pull request merged**